### PR TITLE
LM implementation of multi-wire gates (Mirror of PR#358)

### PR DESCRIPTION
**Pull request by multiphaseCFD**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

This PR add LM implementation for 3 Qubits gates (CSWAP & Toffoli) and QChem 4 Qubits gates (DoubleExcitation, DoubleExcitationMinus, DoubleExcitationPlus).

**Description of the Change:**

Add LM implementation for 3-wires gates (CSWAP & Toffoli) and QChem 4-wires gates (DoubleExcitation, DoubleExcitationMinus, DoubleExcitationPlus), Overloading `revWireParity` for 3 & 4 wires gates.

**Benefits:**

The speed of LM implementation of CSWAP, Toffoli , DoubleExcitation gates, DoubleExcitationMinus and DoubleExcitationPlus  is faster than their PI implementation for all tests number of qubits on Mac M1 chip.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Review by mlxd — APPROVED** [review]
Nothing else to add from me. Thanks @multiphaseCFD

File: pennylane_lightning/src/gates/cpu_kernels/GateImplementationsLM.hpp [hunk 1]
```diff
@@ -66,6 +66,57 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
...
+            fillLeadingOnes(rev_wire[0] + 1) & fillTrailingOnes(rev_wire[1]);
+        parity[2] =
+            fillLeadingOnes(rev_wire[1] + 1) & fillTrailingOnes(rev_wire[2]);
+        parity[3] =
+            fillLeadingOnes(rev_wire[2] + 1) & fillTrailingOnes(rev_wire[3]);
+        parity[4] = fillLeadingOnes(rev_wire[3] + 1);
+
+        return parity;
+    }
+
```
[the remark this answers is not in the document]
Comment (multiphaseCFD) [reply pennylane_lightning/src/gates/cpu_kernels/GateImplementationsLM.hpp:119]: Thanks @mlxd! Good idea! We can do it later.

**Review by chaeyeunpark — APPROVED** [review]
Thanks, @multiphaseCFD, for great work again! I think this PR is ready to be merged.

**Comment by multiphaseCFD** [comment]
> 



> Thanks, @multiphaseCFD, for great work again! I think this PR is ready to be merged.

Thank @chaeyeunpark ! Let me do it!

**Pull request merged**